File: Two pointer/167.Two_Sum_II_Input_Array_Is_Sorted/Two_Sum_II_Input_Array_Is_Sorted.py
from typing import List

class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        ans=[]

        start=0
        end=len(numbers)-1

        while(start<end):
            if(numbers[start]+numbers[end]==target):
                ans.append(start+1)
                ans.append(end+1)
                return ans
            elif(numbers[start]+numbers[end]>target):
                end-=1
            else:
                start+=1
        return ans

if __name__ == "__main__":
    solution = Solution()
    numbers = [2, 7, 11, 15]
    target = 9
    result = solution.twoSum(numbers, target)
    print(result)

File: Two pointer/167.Two_Sum_II_Input_Array_Is_Sorted/Two_Sum_II_Input_Array_Is_Sorted.java

class Solution {

    public int[] twoSum(int[] numbers, int target) {
        int[] ans = new int[2];

        int start = 0;
        int end = numbers.length - 1;

        while (start <= end) {
            if (numbers[start] + numbers[end] == target) {
                ans[0] = start + 1;
                ans[1] = end + 1;
                return ans;
            } else if ((numbers[start] + numbers[end]) > target) {
                end--;
            } else {
                start++;
            }
        }
        return ans;
    }
}

public class Two_Sum_II_Input_Array_Is_Sorted {

    public static void main(String[] args) {
        Solution solution = new Solution();
        int[] numbers = {2, 7, 11, 15};
        int target = 9;
        int[] result = solution.twoSum(numbers, target);
        System.out.println("Index1: " + result[0] + ", Index2: " + result[1]);
    }
}
